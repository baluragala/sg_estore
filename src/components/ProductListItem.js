import React, { Component } from "react";

class ProductListItem extends Component {
  constructor() {
    super();
    this.handleClick = this.handleClick.bind(this);
  }

  handleClick() {
    this.props.onSale(this.props.product.id);
  }

  render() {
    let { title, stock, price } = this.props.product;
    return (
      <div>
        <h2>
          {title}({stock})
        </h2>
        <h2>{price}</h2>
        <button onClick={this.handleClick}>SELL</button>
      </div>
    );
  }
}

export default ProductListItem;
