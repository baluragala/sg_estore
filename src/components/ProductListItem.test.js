import React from "react";
import ProductListItem from "./ProductListItem";
import Enzyme, { shallow } from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import { shallowToJson } from "enzyme-to-json";

Enzyme.configure({ adapter: new Adapter() });

describe("Product List Item", () => {
  let wrapper, product, onSale;
  beforeEach(() => {
    product = { id: 100, title: "p1", price: 20, stock: 20 };
    wrapper = shallow(<ProductListItem product={product} onSale={onSale} />);
    onSale = jest.fn();
  });

  it("should have props", () => {
    console.log(
      wrapper
        .find("h2")
        .at(1)
        .props().children
    );
    expect(wrapper.find("h2").length).toEqual(2);
    expect(
      wrapper
        .find("h2")
        .at(1)
        .props().children
    ).toEqual(20);
    wrapper
      .find("button")
      .first()
      .simulate("click");
    expect(onSale).toBeCalledTimes(1);
    expect(onSale).toBeCalledWith(100);
  });

  it("should have snapshot", () => {
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });
});
