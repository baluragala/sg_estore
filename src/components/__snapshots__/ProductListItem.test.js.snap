// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Product List Item should have snapshot 1`] = `
<div>
  <h2>
    <Link
      replace={false}
      to="/products/100"
    >
      p1
      (
      20
      )
    </Link>
  </h2>
  <h2>
    20
  </h2>
  <button
    onClick={[Function]}
  >
    Sell
  </button>
</div>
`;
